### PR TITLE
Etocs 84 - endpoint to retrieve all journals email alerts from DB

### PR DESCRIPTION
**Pull request by orenard**
The code makes some assumptions about the table names and the attribute names. Should still be easy to refactor if those need changing as I think the main logic is there.

### CODE REVIEW

File: src/main/java/com/bmj/emailalertsapi/constants/package-info.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/**
+ * 
+ */
+/**
+ * @author orenard
+ *
+ */
+package com.bmj.emailalertsapi.constants;
```
Comment (msepler-bmj) [inline src/main/java/com/bmj/emailalertsapi/constants/package-info.java]: Is this package info class necessary?

File: src/main/java/com/bmj/emailalertsapi/controller/package-info.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+/**
+ * 
+ */
+/**
+ * @author orenard
+ *
+ */
+package com.bmj.emailalertsapi.controller;
```
Comment (msepler-bmj) [inline src/main/java/com/bmj/emailalertsapi/controller/package-info.java]: as above

File: src/main/java/com/bmj/emailalertsapi/domain/package-info.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+/**
+ * 
+ */
+/**
+ * @author orenard
+ *
+ */
+package com.bmj.emailalertsapi.domain;
```
Comment (msepler-bmj) [inline src/main/java/com/bmj/emailalertsapi/domain/package-info.java]: as above

File: src/main/java/com/bmj/emailalertsapi/repository/package-info.java [hunk 4]
```diff
@@ -0,0 +1,8 @@
+/**
+ * 
+ */
+/**
+ * @author orenard
+ *
+ */
+package com.bmj.emailalertsapi.repository;
```
Comment (msepler-bmj) [inline src/main/java/com/bmj/emailalertsapi/repository/package-info.java]: as above

File: src/main/java/com/bmj/emailalertsapi/service/package-info.java [hunk 5]
```diff
@@ -0,0 +1,8 @@
+/**
+ * 
+ */
+/**
+ * @author orenard
+ *
+ */
+package com.bmj.emailalertsapi.service;
```
Comment (msepler-bmj) [inline src/main/java/com/bmj/emailalertsapi/service/package-info.java]: as above

File: src/test/java/com/bmj/emailalertsapi/service/JournalsEmailAlertsServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    public void returnEmptyListWhenNoDataInDatabase() {
+        Mockito.when(journalsEmailAlertsRepository.retrieveAllJournalsEmailAlerts())
+               .thenReturn(Collections.emptyList());
+        List<JournalEmailAlert> alertsList = underTest.retrieveAllJournalsEmailAlerts();
+        assertThat(alertsList, Matchers.empty());
+    }
+
+    @Test
+    public void returnThreeAlertsFromDatabase() {
+        // TODO
```
Comment (msepler-bmj) [inline src/test/java/com/bmj/emailalertsapi/service/JournalsEmailAlertsServiceTest.java]: is this test to be implemented later, after the database is available so we can functionally test first?

File: src/main/java/com/bmj/emailalertsapi/constants/package-info.java (hunk 1 above)
Comment (orenard) [reply src/main/java/com/bmj/emailalertsapi/constants/package-info.java]: I'd like to think so. This (https://www.intertech.com/Blog/whats-package-info-java-for/) provides a better answer than I could.

File: src/test/java/com/bmj/emailalertsapi/service/JournalsEmailAlertsServiceTest.java (hunk 6 above)
Comment (orenard) [reply src/test/java/com/bmj/emailalertsapi/service/JournalsEmailAlertsServiceTest.java]: No, I could implement it now and forgot :-). Will do this now.

**Comment by orenard** [comment]
Updates made

File: src/main/java/com/bmj/emailalertsapi/constants/package-info.java (hunk 1 above)
Comment (msepler-bmj) [reply src/main/java/com/bmj/emailalertsapi/constants/package-info.java]: Ah sweet, makes sense! :+1:

**Pull request merged**